app/components/sidebars/settings/settings_sidebar_base.js

```diff
@@ -206,11 +215,27 @@ export default class SettingsSidebarBase extends PureComponent {
     };
 
     renderCustomStatus = () => {
-        const {isCustomStatusEnabled, customStatus, theme} = this.props;
+        const {isCustomStatusEnabled, customStatus, theme, setStatusRequestStatus, clearStatusRequestStatus} = this.props;
         if (!isCustomStatusEnabled) {
             return null;
         }
-        const isStatusSet = customStatus && (customStatus.text || customStatus.emoji);
+
+        let showRetryMessage = false;
+        let isStatusSet = customStatus && customStatus.emoji;
+        if (clearStatusRequestStatus === RequestStatus.STARTED || (clearStatusRequestStatus === RequestStatus.SUCCESS && this.prevClearStatusRequestStatus !== clearStatusRequestStatus)) {
+            isStatusSet = false;
+        }
+        if (clearStatusRequestStatus === RequestStatus.FAILURE && this.prevClearStatusRequestStatus !== clearStatusRequestStatus) {
+            showRetryMessage = true;
+        }
+
+        if (!showRetryMessage && setStatusRequestStatus === RequestStatus.FAILURE && this.prevSetStatusRequestStatus !== setStatusRequestStatus) {
+            showRetryMessage = true;
+        }
+
+        this.prevSetStatusRequestStatus = setStatusRequestStatus;
+        this.prevClearStatusRequestStatus = clearStatusRequestStatus;
```

Comment:
Move this to `componentDidUpdate` instead of `render`

Reply:
Do you mean getDerivedStateFromProps?